media-libs/libavif/libavif-9999.ebuild

```diff
@@ -14,39 +15,61 @@ SLOT="0"
 KEYWORDS=""
 IUSE="+aom dav1d examples extras gdk-pixbuf rav1e svt-av1"
 
-DEPEND="media-libs/libpng
-	sys-libs/zlib
-	virtual/jpeg
-	aom? ( >=media-libs/libaom-2.0.0 )
-	dav1d? ( media-libs/dav1d )
-	gdk-pixbuf? ( x11-libs/gdk-pixbuf:2 )
+DEPEND="media-libs/libpng[${MULTILIB_USEDEP}]
+	sys-libs/zlib[${MULTILIB_USEDEP}]
+	virtual/jpeg[${MULTILIB_USEDEP}]
+	aom? ( >=media-libs/libaom-2.0.0[${MULTILIB_USEDEP}] )
+	dav1d? ( media-libs/dav1d[${MULTILIB_USEDEP}] )
+	gdk-pixbuf? ( x11-libs/gdk-pixbuf:2[${MULTILIB_USEDEP}] )
 	rav1e? ( media-video/rav1e[capi] )
 	svt-av1? ( >=media-libs/svt-av1-0.8.6 )"
 RDEPEND="${DEPEND}"
 BDEPEND="virtual/pkgconfig"
 
 REQUIRED_USE="|| ( aom dav1d )"
 
-src_configure() {
+multilib_src_configure() {
 	local mycmakeargs=(
 		-DBUILD_SHARED_LIBS=ON
 		-DAVIF_CODEC_AOM=$(usex aom ON OFF)
 		-DAVIF_CODEC_DAV1D=$(usex dav1d ON OFF)
 		-DAVIF_CODEC_LIBGAV1=OFF
-		-DAVIF_CODEC_RAV1E=$(usex rav1e ON OFF)
-		-DAVIF_CODEC_SVT=$(usex svt-av1 ON OFF)
 
 		# Use system libraries.
 		-DAVIF_LOCAL_ZLIBPNG=OFF
 		-DAVIF_LOCAL_JPEG=OFF
 
-		-DAVIF_BUILD_EXAMPLES=$(usex examples ON OFF)
-		-DAVIF_BUILD_APPS=$(usex extras ON OFF)
-		-DAVIF_BUILD_TESTS=$(usex extras ON OFF)
-
 		-DAVIF_BUILD_GDK_PIXBUF=$(usex gdk-pixbuf ON OFF)
 	)
 
+	if multilib_is_native_abi; then
+		mycmakeargs+=(
+			-DAVIF_CODEC_RAV1E=$(usex rav1e ON OFF)
+			-DAVIF_CODEC_SVT=$(usex svt-av1 ON OFF)
+
+			-DAVIF_BUILD_EXAMPLES=$(usex examples ON OFF)
+			-DAVIF_BUILD_APPS=$(usex extras ON OFF)
+			-DAVIF_BUILD_TESTS=$(usex extras ON OFF)
+		)
+	else
+		mycmakeargs+=(
+			-DAVIF_CODEC_RAV1E=OFF
+			-DAVIF_CODEC_SVT=OFF
+
+			-DAVIF_BUILD_EXAMPLES=OFF
+			-DAVIF_BUILD_APPS=OFF
+			-DAVIF_BUILD_TESTS=OFF
+		)
+
+		if ! use aom ; then
+			if use rav1e || use svt-av1 ; then
+				ewarn "libavif on $MULTILIB_ABI_FLAG will work in read-only mode."
+				ewarn "Support for rav1e and/or svt-av1 is is not available on $MULTILIB_ABI_FLAG"
+				ewarn "Enable aom flag for full support on $MULTILIB_ABI_FLAG"
```

Comment:
`${VARIABLE}`
https://projects.gentoo.org/qa/policy-guide/ebuild-format.html#pg0101

LGTM, @thesamesam what do you think?

Reply:
Not had a chance to test --- if you can and it works fine, LGTM

Reply:
```
Packages to be tested:
media-libs/libavif-9999

[INFO] >>> Following testing jobs will be executed:
=media-libs/libavif-9999    USE: -aom dav1d examples extras gdk-pixbuf -rav1e svt-av1
=media-libs/libavif-9999    USE: -aom dav1d -examples -extras -gdk-pixbuf -rav1e -svt-av1
=media-libs/libavif-9999    USE: -aom dav1d -examples -extras -gdk-pixbuf rav1e -svt-av1
=media-libs/libavif-9999    USE: aom dav1d -examples extras -gdk-pixbuf rav1e svt-av1
=media-libs/libavif-9999    USE: aom dav1d -examples extras -gdk-pixbuf rav1e -svt-av1
=media-libs/libavif-9999    USE: -aom dav1d -examples extras gdk-pixbuf rav1e -svt-av1
=media-libs/libavif-9999    USE: <default flags>, FEATURES: test
[INFO] >>> Running (1 of 7) =media-libs/libavif-9999 with USE: -aom dav1d examples extras gdk-pixbuf -rav1e svt-av1

These are the packages that would be merged, in order:

Calculating dependencies... done!
[binary  N     ] media-libs/dav1d-0.8.2:0/5::gentoo  USE="10bit 8bit asm" ABI_X86="32 (64) (-x32)" 0 KiB
[binary  N     ] media-libs/svt-av1-0.8.6-1::gentoo  0 KiB
[binary  N     ] media-libs/libjpeg-turbo-2.0.6:0/0.2::gentoo  USE="-java -static-libs" ABI_X86="32 (64) (-x32)" 0 KiB
[binary  N     ] media-libs/libpng-1.6.37-r2-1:0/16::gentoo  USE="apng -static-libs" ABI_X86="32 (64) (-x32)" CPU_FLAGS_X86="sse" 0 KiB
[binary  N     ] virtual/jpeg-100::gentoo  USE="-static-libs" ABI_X86="32 (64) (-x32)" 0 KiB
[binary  N     ] x11-libs/gdk-pixbuf-2.42.2-1:2::gentoo  USE="introspection jpeg -gtk-doc -tiff" ABI_X86="32 (64) (-x32)" 0 KiB
[ebuild  N    *] media-libs/libavif-9999::gentoo  USE="dav1d examples extras gdk-pixbuf svt-av1 -aom -rav1e" ABI_X86="32 (64) (-x32)" 0 KiB

Total: 7 packages (7 new, 6 binaries), Size of downloads: 0 KiB

>>> Verifying ebuild manifests
>>> Emerging binary (1 of 7) media-libs/dav1d-0.8.2::gentoo
>>> Emerging binary (2 of 7) media-libs/svt-av1-0.8.6::gentoo
>>> Emerging binary (3 of 7) media-libs/libjpeg-turbo-2.0.6::gentoo
>>> Emerging binary (4 of 7) media-libs/libpng-1.6.37-r2::gentoo
>>> Installing (1 of 7) media-libs/dav1d-0.8.2::gentoo
>>> Installing (3 of 7) media-libs/libjpeg-turbo-2.0.6::gentoo
>>> Emerging binary (5 of 7) virtual/jpeg-100::gentoo
>>> Installing (4 of 7) media-libs/libpng-1.6.37-r2::gentoo
>>> Installing (2 of 7) media-libs/svt-av1-0.8.6::gentoo
>>> Installing (5 of 7) virtual/jpeg-100::gentoo
>>> Emerging binary (6 of 7) x11-libs/gdk-pixbuf-2.42.2::gentoo
>>> Installing (6 of 7) x11-libs/gdk-pixbuf-2.42.2::gentoo
>>> Emerging (7 of 7) media-libs/libavif-9999::gentoo
>>> Installing (7 of 7) media-libs/libavif-9999::gentoo
>>> Recording media-libs/libavif in "world" favorites file...
>>> Jobs: 7 of 7 complete                           Load avg: 0.55, 0.14, 0.05

 * Messages for package media-libs/libavif-9999:
 * Log file: /var/tmp/portage/vbslogs/build/media-libs/libavif-9999:20210311-081457.log

 * libavif on abi_x86_32 will work in read-only mode.
 * Support for rav1e and/or svt-av1 is is not available on abi_x86_32
 * Enable aom flag for full support on abi_x86_32
 * emktemp is deprecated. Create a temporary file in ${T} instead.
 * emktemp is deprecated. Create a temporary file in ${T} instead.
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.

[INFO] >>> Running (2 of 7) =media-libs/libavif-9999 with USE: -aom dav1d -examples -extras -gdk-pixbuf -rav1e -svt-av1

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R   *] media-libs/libavif-9999::gentoo  USE="dav1d -aom -examples* -extras* -gdk-pixbuf* -rav1e -svt-av1*" ABI_X86="32 (64) (-x32)" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

>>> Verifying ebuild manifests
>>> Emerging (1 of 1) media-libs/libavif-9999::gentoo
>>> Installing (1 of 1) media-libs/libavif-9999::gentoo
>>> Jobs: 1 of 1 complete                           Load avg: 0.62, 0.17, 0.06

 * Messages for package media-libs/libavif-9999:
 * Log file: /var/tmp/portage/vbslogs/build/media-libs/libavif-9999:20210311-081510.log

 * emktemp is deprecated. Create a temporary file in ${T} instead.
 * emktemp is deprecated. Create a temporary file in ${T} instead.

 * Messages for package media-libs/libavif-9999:
 * Log file: /var/tmp/portage/vbslogs/build/media-libs/libavif-9999:20210311-081510.log

 * No AV1 encoder is set,
 * libavif will work in read-only mode.
 * Enable aom, rav1e or svt-av1 flag if you want to save .AVIF files.
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.

[INFO] >>> Running (3 of 7) =media-libs/libavif-9999 with USE: -aom dav1d -examples -extras -gdk-pixbuf rav1e -svt-av1

These are the packages that would be merged, in order:

Calculating dependencies... done!
[binary  N     ] net-libs/libssh2-1.9.0_p20200614::gentoo  USE="zlib -gcrypt (-libressl) -mbedtls" ABI_X86="32 (64) (-x32)" 0 KiB
[binary  N     ] dev-util/cargo-c-0.7.2-1::gentoo  USE="-debug (-libressl)" 0 KiB
[binary  N     ] media-video/rav1e-0.4.0-1::gentoo  USE="capi -debug" 0 KiB
[ebuild   R   *] media-libs/libavif-9999::gentoo  USE="dav1d rav1e* -aom -examples -extras -gdk-pixbuf -svt-av1" ABI_X86="32 (64) (-x32)" 0 KiB

Total: 4 packages (3 new, 1 reinstall, 3 binaries), Size of downloads: 0 KiB

>>> Verifying ebuild manifests
>>> Emerging binary (1 of 4) net-libs/libssh2-1.9.0_p20200614::gentoo
>>> Installing (1 of 4) net-libs/libssh2-1.9.0_p20200614::gentoo
>>> Emerging binary (2 of 4) dev-util/cargo-c-0.7.2::gentoo
>>> Installing (2 of 4) dev-util/cargo-c-0.7.2::gentoo
>>> Emerging binary (3 of 4) media-video/rav1e-0.4.0::gentoo
>>> Installing (3 of 4) media-video/rav1e-0.4.0::gentoo
>>> Emerging (4 of 4) media-libs/libavif-9999::gentoo
>>> Installing (4 of 4) media-libs/libavif-9999::gentoo
>>> Jobs: 4 of 4 complete                           Load avg: 0.97, 0.27, 0.09

 * Messages for package media-libs/libavif-9999:
 * Log file: /var/tmp/portage/vbslogs/build/media-libs/libavif-9999:20210311-081531.log

 * libavif on abi_x86_32 will work in read-only mode.
 * Support for rav1e and/or svt-av1 is is not available on abi_x86_32
 * Enable aom flag for full support on abi_x86_32
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.

[INFO] >>> Running (4 of 7) =media-libs/libavif-9999 with USE: aom dav1d -examples extras -gdk-pixbuf rav1e svt-av1

These are the packages that would be merged, in order:

Calculating dependencies... done!
[binary  N     ] media-libs/libaom-2.0.1:0/2::gentoo  USE="-doc -examples" ABI_X86="32 (64) (-x32)" CPU_FLAGS_X86="avx avx2 mmx sse sse2 sse3 sse4_1 sse4_2 ssse3" 0 KiB
[ebuild   R   *] media-libs/libavif-9999::gentoo  USE="aom* dav1d extras* rav1e svt-av1* -examples -gdk-pixbuf" ABI_X86="32 (64) (-x32)" 0 KiB

Total: 2 packages (1 new, 1 reinstall, 1 binary), Size of downloads: 0 KiB

>>> Verifying ebuild manifests
>>> Emerging binary (1 of 2) media-libs/libaom-2.0.1::gentoo
>>> Installing (1 of 2) media-libs/libaom-2.0.1::gentoo
>>> Emerging (2 of 2) media-libs/libavif-9999::gentoo
>>> Installing (2 of 2) media-libs/libavif-9999::gentoo
>>> Jobs: 2 of 2 complete                           Load avg: 0.98, 0.31, 0.11
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.

[INFO] >>> Running (5 of 7) =media-libs/libavif-9999 with USE: aom dav1d -examples extras -gdk-pixbuf rav1e -svt-av1

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R   *] media-libs/libavif-9999::gentoo  USE="aom dav1d extras rav1e -examples -gdk-pixbuf -svt-av1*" ABI_X86="32 (64) (-x32)" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

>>> Verifying ebuild manifests
>>> Emerging (1 of 1) media-libs/libavif-9999::gentoo
>>> Installing (1 of 1) media-libs/libavif-9999::gentoo
>>> Jobs: 1 of 1 complete                           Load avg: 0.98, 0.33, 0.12
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.

[INFO] >>> Running (6 of 7) =media-libs/libavif-9999 with USE: -aom dav1d -examples extras gdk-pixbuf rav1e -svt-av1

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R   *] media-libs/libavif-9999::gentoo  USE="dav1d extras gdk-pixbuf* rav1e -aom* -examples -svt-av1" ABI_X86="32 (64) (-x32)" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

>>> Verifying ebuild manifests
>>> Emerging (1 of 1) media-libs/libavif-9999::gentoo
>>> Installing (1 of 1) media-libs/libavif-9999::gentoo
>>> Jobs: 1 of 1 complete                           Load avg: 1.07, 0.37, 0.13

 * Messages for package media-libs/libavif-9999:
 * Log file: /var/tmp/portage/vbslogs/build/media-libs/libavif-9999:20210311-081605.log

 * libavif on abi_x86_32 will work in read-only mode.
 * Support for rav1e and/or svt-av1 is is not available on abi_x86_32
 * Enable aom flag for full support on abi_x86_32
 * emktemp is deprecated. Create a temporary file in ${T} instead.
 * emktemp is deprecated. Create a temporary file in ${T} instead.
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.

[INFO] >>> Running (7 of 7) =media-libs/libavif-9999 with USE: <default flags>, FEATURES: test

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R   *] media-libs/libavif-9999::gentoo  USE="aom* -dav1d* -examples -extras* -gdk-pixbuf* -rav1e* -svt-av1" ABI_X86="32 (64) (-x32)" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

>>> Verifying ebuild manifests
>>> Emerging (1 of 1) media-libs/libavif-9999::gentoo
>>> Installing (1 of 1) media-libs/libavif-9999::gentoo
>>> Jobs: 1 of 1 complete                           Load avg: 1.06, 0.39, 0.14

 * Messages for package media-libs/libavif-9999:
 * Log file: /var/tmp/portage/vbslogs/build/media-libs/libavif-9999:20210311-081615.log

 * emktemp is deprecated. Create a temporary file in ${T} instead.
 * emktemp is deprecated. Create a temporary file in ${T} instead.
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.

[INFO] >>> All good.
Grepping through the logs for any errors or QA notices...
Finished grepping.
my-test-container-snap-19843 ~ # grep -r emktemp /var/db/repos/gentoo/profiles/
my-test-container-snap-19843 ~ # grep -r emktemp /var/db/repos/gentoo/eclass/
/var/db/repos/gentoo/eclass/eutils.eclass:# @FUNCTION: emktemp
/var/db/repos/gentoo/eclass/eutils.eclass:emktemp() {
/var/db/repos/gentoo/eclass/eutils.eclass:	eqawarn "emktemp is deprecated. Create a temporary file in \${T} instead."
/var/db/repos/gentoo/eclass/gnome2-utils.eclass:# eutils.eclass: emktemp
/var/db/repos/gentoo/eclass/gnome2-utils.eclass:	local tmp_file=$(emktemp)
my-test-container-snap-19843 ~ # equery f libavif
 * Searching for libavif ...
 * Contents of media-libs/libavif-9999:
/usr
/usr/include
/usr/include/avif
/usr/include/avif/avif.h
/usr/lib
/usr/lib/cmake
/usr/lib/cmake/libavif
/usr/lib/cmake/libavif/libavif-config-gentoo.cmake
/usr/lib/cmake/libavif/libavif-config-version.cmake
/usr/lib/cmake/libavif/libavif-config.cmake
/usr/lib/libavif.so -> libavif.so.10
/usr/lib/libavif.so.10 -> libavif.so.10.0.0
/usr/lib/libavif.so.10.0.0
/usr/lib/pkgconfig
/usr/lib/pkgconfig/libavif.pc
/usr/lib64
/usr/lib64/cmake
/usr/lib64/cmake/libavif
/usr/lib64/cmake/libavif/libavif-config-gentoo.cmake
/usr/lib64/cmake/libavif/libavif-config-version.cmake
/usr/lib64/cmake/libavif/libavif-config.cmake
/usr/lib64/libavif.so -> libavif.so.10
/usr/lib64/libavif.so.10 -> libavif.so.10.0.0
/usr/lib64/libavif.so.10.0.0
/usr/lib64/pkgconfig
/usr/lib64/pkgconfig/libavif.pc
/usr/share
/usr/share/doc
/usr/share/doc/libavif-9999
/usr/share/doc/libavif-9999/README.md.bz2
```
Yep.

Already opened a bug about the *gnome2-utils.eclass* issue.

Reply:
Perfect! Thank you 💯